synapse/handlers/room.py

```diff
@@ -1068,6 +1019,13 @@ async def _send_events_for_new_room(
             A tuple containing the stream ID, event ID and depth of the last
             event sent to the room.
         """
+        visibility = room_config.get("visibility", "private")
+        preset_config = room_config.get(
+            "preset",
+            RoomCreationPreset.PRIVATE_CHAT
+            if visibility == "private"
+            else RoomCreationPreset.PUBLIC_CHAT,
+        )
```

Comment:
I think it would be clearer to put this right above the try-statement on line 1094?